packages/instantsearch-ui-components/src/types/Recommend.ts

```diff
@@ -0,0 +1,27 @@
+export type RecommendClassNames = Partial<{
+  root: string;
+  title: string;
+  container: string;
+  list: string;
+  item: string;
+}>;
+
+export type RecommendTranslations = Partial<{
+  title: string;
+  // Horizontal Slider
+  sliderLabel: string;
+}>;
+
+export type InnerComponentProps<TObject> = {
+  classNames: RecommendClassNames;
+  recommendations: TObject[];
+  translations: RecommendTranslations;
+};
+
+export type RecordWithObjectID<TObject = Record<string, unknown>> = TObject & {
+  objectID: string;
+};
```

Comment:
We have a similar type in [`Hits.tsx`](https://github.com/algolia/instantsearch/blob/55d550eb1030a3b2091917b1a84d280a73a28a3a/packages/instantsearch-ui-components/src/components/Hits.tsx#L7-L9) that we could reuse.